src/porepy/numerics/fv/tpsa.py

```diff
@@ -0,0 +1,1379 @@
+        numbering: _Numbering,
+        filters: _BoundaryFilters,
+        bnd_disp: pp.BoundaryConditionVectorial,
+        boundary_faces: np.ndarray,
+        dist: _Distances,
+    ):
+        """Helper method to construct mappings from cells, and boundary conditions, to
+        faces.
+
+        Parameters:
+            sd: Grid
+            numbering: Container of quantities needed for bookkeeping.
+            filters: Necessary filters for imposing boundary conditions
+            bnd_disp: Boundary condition object for the displacement variable
+            mu_by_dist_fc_cc: The first Lame parameter, mu, divided by the distance
+                between cell and face centers. Expanded to nd.
+
+        Returns:
+            Container of maps from cells to faces.
+
+
+        """
+        nc, nf, nd = sd.num_cells, sd.num_faces, sd.dim
+
+        # Handling of Dirichlet conditions
+        is_dir_nd = bnd_disp.is_dir
+        is_dir = is_dir_nd.ravel("F")
+
+        # Mapping from cell to face, with a weighting of mu / dist. Only interior faces.
+        # Add a factor 2 to compensate for the factor 1/2 included through
+        # multiplication by inv_mu_by_dist_array below; see the construction of
+        # mu_by_dist_array for further comments.
+        cell_to_face = sps.coo_array(
+            ((2 * dist.mu_by_dist_fc_cc, (numbering.fi, numbering.ci))), shape=(nf, nc)
+        ).tocsr()
+
+        # Create the nd version, multiply with a scaling matrix to get an averaging map,
+        # and filter away Dirichlet boundary conditions (these will be enforced
+        # elsewhere in the code).
+        c2f = (
+            filters.dir_nopass_nd
+            @ dist.inv_mu_by_dist_array
+            @ sps.kron(cell_to_face, sps.eye(nd), format="csr")
+        )
+        # Complement map.
+        c2f_compl = sps.csr_matrix(
+            (1 - c2f.data, c2f.indices, c2f.indptr), shape=c2f.shape
+        )
+
+        # Create a mapping for Robin boundary values specifically (note the filter that
+        # lets Robin conditions pass). Inspection of the part of the code where this map
+        # is used will show that Dirichlet and Neumann conditions are treated separately
+        # using filters (which are binary, we cannot do this with Robin since this is a
+        # weighted map).
+        b2f_rob = (
+            filters.rob_pass_nd
+            @ dist.inv_mu_by_dist_array
+            @ sps.dia_array((dist.rob_weight.ravel("F"), 0), shape=(nf * nd, nf * nd))
+        )
+        # For the complement, we only need the diagnoal data.
+        b2f_rob_compl = 1 - b2f_rob.diagonal()
+
+        # Map from scalar cell quantities to nd face quantities (used e.g. for the solid
+        # pressure).
+        c2f_scalar_2_nd = dist.inv_mu_by_dist_array @ sps.kron(
+            cell_to_face, sps.csr_array(np.ones((nd, 1))), format="csr"
+        )
+        # Specifically set a zero value for faces that have Dirichlet conditions, as
```

Comment:
I think the chances of them making such a change are slim, and I am confident it will result in failure of tests. Other implementations must be possible, but I don't see an approach that will not be much more cumbersome. So in total, though I agree you have a good point, I am comfortable with keeping this approach.